corehq/apps/auditcare/models.py

```diff
@@ -53,12 +49,27 @@ def getdate():
 ]
 
 
-class AuditEvent(Document):
-    user = StringProperty()  # the user committing the action
-    base_type = StringProperty(default="AuditEvent")  # for subclassing this needs to stay consistent
-    # subclasses will be known directly from the doc_type, so it's not here.
-    event_date = DateTimeProperty(default=getdate)
-    description = StringProperty()  # particular instance details of this audit event
+class UserAgent(models.Model):
+    value = models.CharField(max_length=255, db_index=True)
+
+
+class HttpAccept(models.Model):
+    value = models.CharField(max_length=255, db_index=True)
+
+
+class ViewName(models.Model):
+    value = models.CharField(max_length=255, db_index=True)
```

Comment:
I don't think we care about having only unique values as long as duplicates don't occur very often. For example, there is a race condition in `get_or_create()` running on multiple hosts, and it's quite possible that more than one host could create a new record with the same value. Error handling for unique constraint violations seems like it would add extra complexity. I think we'd need a custom implementation of `get_or_create`.

As for making `value` the primary key, we'd still need an integer ID field since the whole point of storing the values in a separate table is to have an integer reference for long values. In addition to the issue already mentioned with unique constraints, we'd need a custom implementation of the lookup code since we could no longer use `ForeignKey` to maintain the reference.

> we're currently ignoring the primary keys entirely

They're used when reading auditcare data, which admittedly is a rare use case.

Reply:
I'll leave it to your judgment, but I think ignoring the unique constraint introduces more complexity than simply handling the error. That race-condition error seems straight-forward to handle -- `get_or_create` and if we catch the error code that indicates a duplicate record was found, we just do another query and return the duplicate record.  That can be nicely encapsulated behind a function. Conversely, not handling it uniquely adds another burden of knowledge to our codebase. It would be punishing users who (correctly) prefer `get` over `filter`, thinking the tables are structured in the way we'd expect.

Agree with the point about not using them as a primary key, however.

Reply:
This turned out to be much worse than I had anticipated. Fixes in https://github.com/dimagi/commcare-hq/pull/29391 and https://github.com/dimagi/commcare-hq/pull/29393